pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Validator interface {
+	IsValid(k futureKey, r *validator) (bool, error)
+}
+
+// GitHubValidator.IsValid skips visiting all github issue/PR links.
+func (v GitHubValidator) IsValid(k futureKey, r *validator) (bool, error) {
+	// Find rightmost index of match i.e, where regex match ends.
+	// This will be where issue/PR number starts. Split incase of section link and convert to int.
+	rightmostIndex := v._regex.FindStringIndex(k.dest)
+	stringNum := strings.Split(k.dest[rightmostIndex[1]:], "#")
+	num, err := strconv.Atoi(stringNum[0])
+	if err != nil {
+		return false, err
+	}
+	// If number in link does not exceed then link is valid.
+	if v._maxNum >= num {
+		return true, nil
+	}
+	return false, nil
+}
+
+// RoundTripValidator.IsValid returns true if url is checked by colly.
+func (v RoundTripValidator) IsValid(k futureKey, r *validator) (bool, error) {
+	// Result will be in future.
+	r.destFutures[k].resultFn = func() error { return r.remoteLinks[k.dest] }
+	r.rMu.RLock()
+	if _, ok := r.remoteLinks[k.dest]; ok {
+		r.rMu.RUnlock()
+		return true, nil
+	}
+	r.rMu.RUnlock()
+
+	r.rMu.Lock()
+	defer r.rMu.Unlock()
+	// We need to check again here to avoid race.
+	if _, ok := r.remoteLinks[k.dest]; ok {
```

Comment:
This is some sort of "caching" that we can use on TOP of validators, so other validators can leverage it no? It's ok to think about it later PRs (: